### PR TITLE
Support adding trust bundles to data place HTTP clients

### PR DESCRIPTION
**Pull request by pierDipi**
This is the data plane only part related to the trust-manager integration: https://github.com/knative/eventing/pull/7532

- Add trustBundles to contract ConfigMap
- Detect trust bundle changes and re-create HTTP clients

### CODE REVIEW

**Review by Cali0707 — COMMENTED** [review]
This looks good overall, I just had a question about one small part of the code I didn't understand

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/file/FileWatcher.java [hunk 1]
```diff
@@ -113,6 +119,8 @@ public void run() {
         }
         logger.info("Started watching {}", toWatch);
 
+        this.waitRunning.countDown();
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/file/FileWatcher.java:122]: This is to ensure that tests only write to the file when the watcher is actually running

**Review by Cali0707 — APPROVED** [review]
/lgtm

Thanks for the explanation on the bits I didn't understand @pierDipi !

**Pull request merged**